onnxruntime/wasm/api.cc

```diff
@@ -387,3 +395,102 @@ char* OrtEndProfiling(ort_session_handle_t session) {
              ? file_name
              : nullptr;
 }
+
+// Training API Section
+
+#ifdef ENABLE_TRAINING_APIS
+#define CHECK_TRAINING_STATUS(ORT_API_NAME, ...) \
+  CheckStatus(Ort::GetTrainingApi().ORT_API_NAME(__VA_ARGS__))
+
+ort_training_checkpoint_handle_t EMSCRIPTEN_KEEPALIVE OrtTrainingLoadCheckpointState(void* checkpoint_data_buffer, size_t checkpoint_size) {
+  OrtCheckpointState* checkpoint_state = nullptr;
+  return (CHECK_TRAINING_STATUS(LoadCheckpointFromBuffer, checkpoint_data_buffer, checkpoint_size, &checkpoint_state) == ORT_OK)
+             ? checkpoint_state
+             : nullptr;
+}
+
+void EMSCRIPTEN_KEEPALIVE OrtTrainingReleaseCheckpoint(ort_training_checkpoint_handle_t training_checkpoint_state_handle) {
+  Ort::GetTrainingApi().ReleaseCheckpointState(training_checkpoint_state_handle);
+}
+
+ort_training_session_handle_t EMSCRIPTEN_KEEPALIVE OrtTrainingCreateSession(const ort_session_options_handle_t options,
+                                                                            ort_training_checkpoint_handle_t training_checkpoint_state_handle,
+                                                                            void* train_model,
+                                                                            size_t train_size,
+                                                                            void* eval_model,
+                                                                            size_t eval_size,
+                                                                            void* optimizer_model,
+                                                                            size_t optimizer_size) {
+  OrtTrainingSession* training_session = nullptr;
+  return (CHECK_TRAINING_STATUS(CreateTrainingSessionFromArray, g_env, options,
+                                training_checkpoint_state_handle, train_model, train_size,
+                                eval_model, eval_size, optimizer_model, optimizer_size,
+                                &training_session) == ORT_OK)
+             ? training_session
+             : nullptr;
+}
+
+int EMSCRIPTEN_KEEPALIVE OrtTrainingLazyResetGrad(ort_training_session_handle_t training_handle) {
+  return CHECK_TRAINING_STATUS(LazyResetGrad, training_handle);
+}
+
+ort_tensor_handle_t EMSCRIPTEN_KEEPALIVE OrtTrainingRunTrainStep(ort_training_session_handle_t training_handle,
```

Comment:
The header says this function returns error code but in implementation it returns a tensor handle.

Reply:
Oops, fixed in latest commit